cms/models/pluginmodel.py

```diff
@@ -292,9 +292,8 @@ def copy_plugin(self, target_placeholder, target_language, parent_cache, no_sign
             new_plugin._no_reorder = True
         new_plugin.save()
         if plugin_instance:
-            if plugin_instance.__class__ == CMSPlugin:
-                # get a new instance so references do not get mixed up
-                plugin_instance = CMSPlugin.objects.get(pk=plugin_instance.pk)
+            # get a new instance so references do not get mixed up
+            plugin_instance = plugin_instance.__class__.objects.get(pk=plugin_instance.pk)
             plugin_instance.pk = new_plugin.pk
             plugin_instance.id = new_plugin.pk
```

Comment:
I'll shut up about this now. Seeing this just confused me as it doesn't feel right and I hoped that because you touched code near it maybe you'd have some insight as to what's going on.